frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripServlet.java

```diff
@@ -0,0 +1,63 @@
+package com.google.tripmeout.frontend.servlet;
+
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+import com.google.tripmeout.frontend.storage.TripStorage;
+import java.io.IOException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Singleton
+public final class TripServlet extends HttpServlet {
+  // The Trip get and delete servlet
+  private static final long serialVersionUID = 1L;
+  
+  private static final String APPLICATION_JSON_CONTENT_TYPE = "application/json;";
+  
+  private final Gson gson;
+  private final TripStorage storage;
+
+  @Inject
+  public TripServlet( TripStorage storage, Gson gson) {
+    this.storage = storage;
+    this.gson = gson;
+  }
+
+  @Override
+  public void doGet(final HttpServletRequest request, final HttpServletResponse response)
+      throws IOException {
+    try {
+      final String tripId = TripName.fromRequestUri(request.getRequestURI()).id();
```

Comment:
Store the `TripName` itself here and call `storage.getTrip(tripName.id())` below.

This is a nit, but I want to take some time to explain. The idea behind using TripName is to give ourselves stronger typing and more structure around the resource name itself. To take advantage of this, the best practice is to hang onto the strongly-typed `TripName` for as long as possible and only access the stringly-typed ID at the last minute possible.

In some cases, it might even make sense to have APIs take/return this named type explicitly. Doing so can prevent accidentally swapping the wrong value in a String (or other general type) parameter. I won't recommend that here since that would couple the REST concerns (API endpoint names) with our storage API, but the general pattern is worth considering.